crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -110,6 +133,63 @@ impl C8yMapperActor {
         Ok(())
     }
 
+    /// Performs auto-registration process for an entity under a given
+    /// identifier.
+    ///
+    /// If an entity is a service, its device is also auto-registered if it's
+    /// not already registered.
+    ///
+    /// It returns MQTT register messages for the given entities to be published
+    /// by the mapper, so other components can also be aware of a new device
+    /// being registered.
+    fn auto_register_entity(&mut self, entity_id: &str) -> Vec<Message> {
+        let mut register_messages = vec![];
+        if let Some(("te", topic)) = entity_id.split_once('/') {
+            let (device_id, service_id) = match topic.split('/').collect::<Vec<&str>>()[..] {
+                ["device", device_id, "service", service_id, ..] => (device_id, Some(service_id)),
+                ["device", device_id, ..] => (device_id, None),
```

Comment:
I would discard any message that strictly follow the default scheme.

```suggestion
                ["device", device_id, "", ""] => (device_id, None),
```